src/pybind/mgr/dashboard/frontend/src/app/shared/components/grafana/grafana.component.ts

```diff
@@ -163,19 +167,31 @@ export class GrafanaComponent implements OnInit, OnChanges {
   }
 
   getFrame() {
-    this.settingsService
-      .validateGrafanaDashboardUrl(this.uid)
-      .subscribe((data: any) => (this.dashboardExist = data === 200));
-    this.url =
-      this.baseUrl +
-      this.uid +
-      '/' +
-      this.grafanaPath +
-      '&refresh=2s' +
-      `&var-datasource=${this.datasource}` +
-      this.mode +
-      '&' +
-      this.time;
+    if (this.type === 'metrics') {
+      this.settingsService
+        .validateGrafanaDashboardUrl(this.uid)
+        .subscribe((data: any) => (this.dashboardExist = data === 200));
+      this.url =
+        this.baseUrl +
+        this.uid +
+        '/' +
+        this.grafanaPath +
+        '&refresh=2s' +
+        `&var-datasource=${this.datasource}` +
+        this.mode +
+        '&' +
+        this.time;
+    } else {
+      this.baseUrl = this.baseUrl.slice(0, -2);
+      this.url =
+        this.baseUrl +
+        this.grafanaPath +
+        'orgId=1' +
+        '&left=["now-1h","now",' +
+        `"${this.datasource}",` +
+        '{"refId":"A"}]' +
+        '&kiosk=tv';
+    }
```

Comment:
Assuming `metrics` are the existing behaviour wouldn't it be easier to create a condition like
```
if (this.type === 'logs') 
 .... 
 ```
 
 and put the new functionality in it. And put the old functionality in the else case?

Reply:
Seems to be one and the same thing.